docs: upgrade docs now reflect the new upgrade commands (#1331)
edgelesssys/constellation: docs/docs/workflows/upgrade.md

# Upgrade your cluster

> Constellation provides an easy way to upgrade all components of your cluster, without disrupting it's availability.
> Specifically you can upgrade the Kubernetes version, the nodes' image and the Constellation microservices.

Reviewer: ```suggestion
Constellation provides an easy way to upgrade all components of your cluster, without disrupting its availability.
Specifically, you can upgrade the Kubernetes version, the nodes' image, and the Constellation microservices.
```
MS style guide recommends a comma following introductory phrases and also the Oxford comma

You configure the desired versions in your local Constellation configuration and trigger upgrades with the `upgrade apply` command.
To learn about available versions you use the `upgrade check` command.
Which versions are available depends on the CLI version you are using.

## Update the CLI

> Each CLI comes with a set of microservice and Kubernetes versions
> It's recommend to always use the most up-to-date CLI.
> However, there are two constraints to this recommendation:
> 1. Don't upgrade more than one minor version at a time.
> For example, if you are currently on CLI version v2.6, you should upgrade to v2.7, upgrade your cluster and only then continue upgrading the CLI after.
> 2. Don't upgrade to a CLI version that doesn't support your current Kubernetes version.
> If your current Kubernetes version is dropped from the next CLI version, use your current CLI to upgrade to a newer Kubernetes version first.

Reviewer: I find this hard to understand. I tried rephrasing it to what you should do instead of what you shouldn't do:
```suggestion
Each CLI comes with a set of supported microservice and Kubernetes versions.
Most importantly, a given CLI version can only upgrade a cluster of the previous minor version, but not older ones.
This means that you have to upgrade your CLI and cluster one minor version at a time.

For example, if you are currently on CLI version v2.6 and the latest one is v2.8, you should

* upgrade the CLI to v2.7,
* upgrade the cluster to v2.7,
* and only then continue upgrading the CLI (and the cluster) to v2.8 after.

Also note that if your current Kubernetes version isn't supported anymore by the next CLI version, use your current CLI to upgrade to a newer Kubernetes version first.
```
What do you think?

Author: I like it a lot. Changed it.

Each CLI supports at least three Kubernetes versions.

> To learn which Kubernetes versions are supported by a particular CLI run [constellation config kubernetes-versions](../reference/cli.md#constellation-config-kubernetes-versions).

Reviewer: ```suggestion
To learn which Kubernetes versions are supported by a particular CLI, run [constellation config kubernetes-versions](../reference/cli.md#constellation-config-kubernetes-versions).
```

## Migrate the configuration

The Constellation configuration file is located in the file `constellation-conf.yaml` in your workspace.
Refer to the [migration reference](../reference/config-migration.md) to check if you need to update fields in your configuration file.

## Check for upgrades

> To learn which versions the current CLI can upgrade to and what's installed in your cluster use this command to see available upgrades:

Reviewer: ```suggestion
To learn which versions the current CLI can upgrade to and what's installed in your cluster, run:
```

```bash
constellation upgrade check
```

You can either enter the reported target versions into your config manually or run the above command with the `--write-config` flag.

> When using this flag the `kubernetesVersion`, `image` and `microserviceVersion` fields are overwritten with the smallest available upgrade.

Reviewer: ```suggestion
When using this flag, the `kubernetesVersion`, `image`, and `microserviceVersion` fields are overwritten with the smallest available upgrade.
```

## Apply the upgrade

> Once you updated your config with the versions you want in your cluster, you can trigger the upgrade with this command:

Reviewer: ```suggestion
Once you updated your config with the desired versions, you can trigger the upgrade with this command:
```

```bash
constellation upgrade apply
```

Microservice upgrades will be finished within a few minutes, depending on the cluster size.

> If you are interested, you can monitor pods restarting in the `kube-system` namespace with your tool of choice.
> Image and Kubernetes upgrades will take longer.
> This is because for each node in your cluster, a new node has to be created and joined.
> The process usually takes ~6 minutes per node.

Reviewer: ```suggestion
If you are interested, you can monitor pods restarting in the `kube-system` namespace with your tool of choice.

Image and Kubernetes upgrades take longer.
For each node in your cluster, a new node has to be created and joined.
The process usually takes about 6 minutes per node.
```
I'd start a new paragraph (and simplify the wording a bit).

6 minutes sound like a very accurate timing. Is this really true, independent of CSP or node type? (If not, maybe replace with sth like "about five minutes" or "up to ten minutes"?)
